tests/passes/dead_code_elimination_test.py

```diff
@@ -266,6 +266,54 @@ def nested(a: dace.float64[20], b: dace.float64[20]):
     sdfg.validate()
 
 
+def test_dde_inout_two_states():
+    """Test two states with read/write in second state."""
+
+    sdfg = dace.SDFG("dde_inout_two_states")
+    sdfg.add_scalar("tmp", dace.float32)
+    sdfg.add_scalar("computed", dace.float32, transient=True)
+
+    start_state = sdfg.add_state("start_state", is_start_block=True)
+    s1_read_tmp = start_state.add_read("tmp")
+    s1_write_computed = start_state.add_write("computed")
+
+    # upstream tasklet that writes a transient (to be read in a separate state)
+    first_tasklet = start_state.add_tasklet("write", {"read_tmp"}, {"write_computed"},
+                                            "write_computed = read_tmp * 2 + 1")
+    start_state.add_memlet_path(s1_read_tmp, first_tasklet, dst_conn="read_tmp", memlet=dace.Memlet(data="tmp"))
+    start_state.add_memlet_path(first_tasklet,
+                                s1_write_computed,
+                                src_conn="write_computed",
+                                memlet=dace.Memlet(data="computed"))
+
+    next_state = sdfg.add_state_after(start_state, "next_state")
+    s2_write_computed = next_state.add_write("computed")
+    s2_read_computed = next_state.add_read("computed")
+    s2_write_tmp = next_state.add_write("tmp")
+
+    # downstream tasklet that reads _and_ writes a transient
+    second_tasklet = next_state.add_tasklet(
+        "read_write", {"read_computed"}, {"write_tmp", "write_computed"},
+        "write_computed = 2 * read_computed\nwrite_tmp = write_computed + read_computed")
+    next_state.add_memlet_path(s2_read_computed,
+                               second_tasklet,
+                               dst_conn="read_computed",
+                               memlet=dace.Memlet(data="computed"))
+    next_state.add_memlet_path(second_tasklet, s2_write_tmp, src_conn="write_tmp", memlet=dace.Memlet(data="tmp"))
+    next_state.add_memlet_path(second_tasklet,
+                               s2_write_computed,
+                               src_conn="write_computed",
+                               memlet=dace.Memlet(data="computed"))
+
+    results = {}
+    Pipeline([DeadDataflowElimination()]).apply_pass(sdfg, results)
+
+    dde_results = results["DeadDataflowElimination"]
```

Comment:
```suggestion
    dde_results = results["DeadDataflowElimination"][0]
```

The structure of pipeline results on `main` is slightly different compared to `v1/maintenance`.

Can't push to your fork, but this is all it should (TM) take to get the "General Tests" green again. Not sure about the FPGA tests. That seems odd and might hint that this fix is too simplistic.

Reply:
Let's push this to begin with